cumulus/polkadot-parachain/src/common/solochain.rs

```diff
@@ -0,0 +1,263 @@
+			sc_service::build_network(sc_service::BuildNetworkParams {
+				config: &config,
+				client: client.clone(),
+				transaction_pool: transaction_pool.clone(),
+				spawn_handle: task_manager.spawn_handle(),
+				import_queue,
+				net_config,
+				block_announce_validator_builder: None,
+				warp_sync_params: None,
+				block_relay: None,
+				metrics,
+			})?;
+
+		if config.offchain_worker.enabled {
+			task_manager.spawn_handle().spawn(
+				"offchain-workers-runner",
+				"offchain-worker",
+				sc_offchain::OffchainWorkers::new(sc_offchain::OffchainWorkerOptions {
+					runtime_api_provider: client.clone(),
+					is_validator: config.role.is_authority(),
+					keystore: Some(keystore_container.keystore()),
+					offchain_db: backend.offchain_storage(),
+					transaction_pool: Some(OffchainTransactionPoolFactory::new(
+						transaction_pool.clone(),
+					)),
+					network_provider: Arc::new(network.clone()),
+					enable_http_requests: true,
+					custom_extensions: |_| vec![],
+				})
+				.run(client.clone(), task_manager.spawn_handle())
+				.boxed(),
+			);
+		}
+
+		let rpc_extensions_builder = {
+			let client = client.clone();
+			let pool = transaction_pool.clone();
+
+			Box::new(move |deny_unsafe, _| {
+				let mut module = RpcModule::new(());
+				module
+					.merge(System::new(client.clone(), pool.clone(), deny_unsafe).into_rpc())
+					.map_err(|e| ServiceError::Application(e.into()))?;
+				Ok(module)
+			})
+		};
+
+		let prometheus_registry = config.prometheus_registry().cloned();
+
+		sc_service::spawn_tasks(sc_service::SpawnTasksParams {
+			config,
+			client: client.clone(),
+			backend,
+			task_manager: &mut task_manager,
+			keystore: keystore_container.keystore(),
+			transaction_pool: transaction_pool.clone(),
+			rpc_builder: rpc_extensions_builder,
+			network,
+			system_rpc_tx,
+			tx_handler_controller,
+			sync_service,
+			telemetry: telemetry.as_mut(),
+		})?;
+
+		let proposer = sc_basic_authorship::ProposerFactory::new(
+			task_manager.spawn_handle(),
+			client.clone(),
+			transaction_pool.clone(),
+			prometheus_registry.as_ref(),
+			telemetry.as_ref().map(|x| x.handle()),
+		);
+
+		// Start ManualSeal consensus
+		let (mut sink, commands_stream) = futures::channel::mpsc::channel(1024);
+		task_manager.spawn_handle().spawn("block_authoring", None, async move {
+			loop {
+				futures_timer::Delay::new(std::time::Duration::from_millis(block_time)).await;
+				sink.try_send(sc_consensus_manual_seal::EngineCommand::SealNewBlock {
+					create_empty: true,
+					finalize: true,
+					parent_hash: None,
+					sender: None,
+				})
+				.unwrap();
+			}
+		});
+
+		let params = sc_consensus_manual_seal::ManualSealParams {
+			block_import: client.clone(),
+			env: proposer,
+			client,
+			pool: transaction_pool,
+			select_chain,
+			commands_stream: Box::pin(commands_stream),
+			consensus_data_provider: None,
+			create_inherent_data_providers: move |_, ()| async move {
```

Comment:
This is tied to the runtime, as in it is also part of the `assumptions` and need to ideally be tested for. For example, will this work with a runtime that doesn't have pallet-timestamp? 

@gupnik when I talk about "opinionated frame system super pallet", this is the type of things that I mean. All normal FRAME-based runtimes must have the time timestamp pallet.   Why should this not already be part of `frame_system`, or an opinionated flavor of it? 

A project that you can generally start to complement the omni-node is an `frame-omni-node-system`, a variant of `frame_system` and defaults that is meant to work with omni-node. 

cc @shawntabrizi